src/main/java/com/empowerops/linqalike/delegate/BiCondition.java

```diff
@@ -0,0 +1,13 @@
+package com.empowerops.linqalike.delegate;
+
+import com.empowerops.linqalike.common.Tuple;
+
+@FunctionalInterface
+public interface BiCondition<TFirstInspected, TSecondInspected> {
+
+    public boolean passesFor(TFirstInspected left, TSecondInspected right);
+
+    default Condition<Tuple<? extends TFirstInspected, ? extends TSecondInspected>> toConditionOnTuple(){
+        return tuple -> this.passesFor(tuple.left, tuple.right);
+    }
```

Comment:
I wonder if I should have `BiCondition` simply implement `Condition<Tuple>`? saves me the factory call, but it makes the type inference engine's life harder.